### DISCUSSION TITLE
Add support for using the authenticated user within GraphQL queries or mutations

### DISCUSSION THREAD

**Discussion by jesstelford** [keystonejs/keystone#7248] [Feature requests]
I'm building an events platform where a `User` can belong to multiple `Communities`.
I want to get a list of communities the user _hasn't yet_ joined.
Currently, I have to make **TWO queries**:
1. Get the user's ID
2. Query for communities with `members: { none: { user: { id: $userId } } }`.

Ideally, I'd like to do a single query based on the currently authenticated User, something like:

```graphql
query {
  communities(where: { members: { none: { user: { authenticatedUser: true } } } }) {
    id
  }
}
```

Another, more complex example: `Communities` can host `Events` that the user can `RSVP` to. I want to get a list of events the user _hasn't yet_ RSVP'd to, limited to the communities they're already a member of:

```graphql
query {
  authenticatedUser {
    communities {
      id
      events(where: { rsvps: { none: { user: { authenticatedUser: true } } } }) {
        id
      }
    }
  }
}
```

The `user: { authenticatedUser: true }` would be a filter available anywhere there is a `User` relationship (including the root level `users` query), and would internally use the authenticated item's `id` as the filter (from the session).
When there's no currently authenticated user, I'm not sure what should happen. Perhaps one or both of these options are valid:

1. Return an error
2. Map `authenticatedUser: true` to `id: -1` (or some other impossible ID, so it'll _never_ match).

**Comment by jesstelford** [comment]
In the mean-time, is there a workaround to avoid the double-query problem? Perhaps using virtual fields or some such?

**Reply by jesstelford** [reply]
Looks like we [can't use](https://keystonejs.com/docs/apis/filters#virtual) Virtual fields:

> **Virtual type**
> The virtual field type does not support filters.

**Reply by jesstelford** [reply]
Maybe a [Custom Field](https://keystonejs.com/docs/guides/custom-fields) which looks like a checkbox on the GraphQL side, but acts as an ID filter on the Prisma side? Anyone know how we might build something like that?

**Reply by dcousens** [reply]
If a work-around is posted,  it might be helpful to update https://github.com/keystonejs/keystone/discussions/7514

**Reply by christopherliu** [reply]
I had that similar issue above (#7514) and it looks like the [Access Control API](https://keystonejs.com/docs/apis/access-control) can prevent the security hole that comes with allowing random access, though I'd still have to insert the user ID manually.
